controllers/halqahController.js

```diff
@@ -1,64 +1,52 @@
 import Halqah from '../models/halqahModel.js';
 import asyncHandler from '../middleware/asyncHandler.js';
-const createHalqah = async (req, res) => {
+
+const createHalqah = asyncHandler(async (req, res, next ) => {
+
   const newHalqah = await Halqah.create(req.body);
   res.status(201).json({
     status: 'success',
     data: {
       message: newHalqah,
     },
   });
-};
+}) 
 
-const getAllHalaqaat = asyncHandler(async (req, res, next) => {
-  // try {
-  //   let query;
-  //   const requestQuery = {...req.query}
 
-  //remove fields
-  //const removedFields = ['select']
-  // removedFields.forEach(param => delete requestQuery[param])
-  // console.log(removedFields)
+const getAllHalaqaat = asyncHandler(async (req, res, next ) => {
 
-  // if(req.query.select){
-  //   const fields = req.query.select.split(',').join(' ')
-  //   query = query.select(fields)
-  // }
-  const halqah = await Halqah.find({});
+  const { page = 1, limit = 10 } = req.query
+
+  const halqah = await Halqah.find({}).skip((limit * page) - limit).limit(limit * 1);
   res.status(200).json({
     status: 'success',
     data: { halqah },
   });
 });
 
-const getHalqah = async (req, res, next) => {
-  try {
-    const halqah = await Halqah.findById(req.params.id);
+const getHalqah = asyncHandler( async (req, res, next ) => {
+
+  const { id } = req.params
+  const halqah = await Halqah.findById( id );
+  res.status(200).json({
+    status: 'success',
+    data: { halqah },
+  });
+}); 
+
+
+const updateHalqah = asyncHandler( async (req, res, next ) => {
 
-    res.status(200).json({
-      status: 'success',
-      data: {
-        halqah,
-      },
-    });
-  } catch (err) {
-    console.log(err);
-  }
-};
+  const { params: { id }, body } = req
```

Comment:
This can work when I'm changing the whole body. What if I want to change one of those fields.

Reply:
Done 
I have used the PUT verb in the halqahRoute